src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -123,6 +122,13 @@ private static string NormalizePath(string path)
         /// </returns>
         private static string GetCorrectCasedPath(string path)
         {
+            path = NormalizePath(path);
+
+            if (Platform.IsLinux)
+            {
+                return path;
+            }
```

Comment:
Aren't MacOS paths also case sensitive? Is there another reason we'd have this behaviour for Linux but not MacOS?

Reply:
By default MacOS uses a case-insensitive file system.

Reply:
The primary scenario where we hit a problem is with git where the remote repo is on Linux and case-sensitive and someone adds a file with a directory that they manually typed and then you have two directories in git that only differ by case.  I think that affects Windows and macOS clients even if their local filesystems are case insensitive.

Reply:
If git is case-sensetive locally on all platforms then users should follow the application design why should PowerShell do expensive work for all paths and for other scenarios?

Reply:
>  they manually typed 

> the remote repo is on Linux and case-sensitive 

Wow! If repo/application is case-sensitive and user obviously know this shouldn't the user take care of compliance himself?

If different platforms have different behavior, this is a fundamental and unsolvable problem. We constantly get requests about slashes because they are different in local (Windows) and remote session (Linux). I haven't seen as many requests about case, but the root of the problem is the same.
It should be clearly written in the documentation that this is the user's concern: if a scenario is  - scripts works on Windows client and remote sessions are on Linux-s (and vise-versa) a recommendation for users should be to prefer forward slashes (backslashs) and be careful about cases.